packages/svelte/src/compiler/phases/3-transform/client/utils.js

```diff
@@ -639,3 +639,13 @@ export function with_loc(target, source) {
 	}
 	return target;
 }
+
+/**
+ * @param {string} filename
+ */
+export function hash_filename(filename) {
+	for (var h = 0, i = 0; i < filename.length; h &= h) {
+		h = 31 * h + filename.charCodeAt(i++);
+	}
+	return h;
+}
```

Comment:
we have a `hash` function already in the codebase (in `2-analyze/utils.js`; it would need to be moved if it were to be used here too), is there a reason we can't reuse it here?